libqtile/widget/net.py

```diff
@@ -29,43 +29,45 @@ class Net(base.ThreadedPollText):
     """Displays interface down and up speed"""
     orientations = base.ORIENTATION_HORIZONTAL
     defaults = [
-        ('interface', 'wlan0', 'The interface to monitor'),
+        ('interface', None, 'List of interfaces to monitor or None to displays all active NICs combined'),
         ('update_interval', 1, 'The update interval.'),
     ]
 
     def __init__(self, **config):
         base.ThreadedPollText.__init__(self, **config)
         self.add_defaults(Net.defaults)
-        self.interfaces = self.get_stats()
+        if not isinstance(self.interface, list):
+            if self.interface is None:
+                self.interface = "all"
+            self.interface = [self.interface]
+        self.stats = self.get_stats()
 
     def convert_b(self, b):
-        # Here we round to 1000 instead of 1024
-        # because of round things
-        letter = 'B'
-        # b is a float, so don't use integer division
-        if int(b / 1000) > 0:
-            b /= 1000.0
-            letter = 'k'
-        if int(b / 1000) > 0:
-            b /= 1000.0
-            letter = 'M'
-        if int(b / 1000) > 0:
-            b /= 1000.0
-            letter = 'G'
-        # I hope no one have more than 999 GB/s
-        return b, letter
 
-    def get_stats(self):
+        letters = ["b", "kb", "Mb", "Gb", "Tb", "Pb", "Eb", "Zb", "Yb"]
+        factor = 1000.0
+        i = 0
 
-        interfaces = {}
-        net = psutil.net_io_counters(pernic=True)
-        for iface in net:
-            name = iface
-            down = net[iface].bytes_recv
-            up = net[iface].bytes_sent
-            interfaces[name] = {'down': down, 'up': up}
+        while int(b / 1000) > 0:
+            b /= factor
```

Comment:
This should probably check as well that `i < len(letters) - 1`, in case some erroneous case happens where over 1 Yb is reported, at least we won't crash the widget. This should probably also read `b < factor` rather than `int(b / 1000) > 0`.

I'm wondering if we want to get fancy with this. Something like:
```
idx = log(b) / log(1000)
```
to compute the index of the letter. One additional thing that would be needed here is to be sure 0 and (erroneous) negative numbers aren't fed into the log, so maybe that isn't ideal.

Reply:
This is what I use in another project:

```python
chosen_unit = "B"
for unit in ("KiB", "MiB", "GiB", "TiB"):
    if value > 1000:
        value /= 1024
        chosen_unit = unit
    else:
        break
```

It think it handles the two cases you mentioned. My 2 cents :slightly_smiling_face:

Reply:
May I propose:
```python
def convert_b(self, b):

    letters = ["kB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"]
    factor = 1000.0
    unit = "B"

    for letter in letters:
        if b > factor:
            b /= factor
            unit = letter
        else:
            break
    return b, unit
```
I'm in favor of a solution based on @pawamoy's comment:
* cleaner
* no extra check to make sure the values don't escalate too quickly
* no while loop
* I don't think there's much more to optimize